lib/hooks.js

```diff
@@ -52,7 +52,7 @@ Hooks.prototype.validate = function (hook, fn) {
   if (supportedHooks.indexOf(hook) === -1) {
     throw new Error(`${hook} hook not supported!`)
   }
-  if (typeof fn !== 'function') throw new FST_ERR_HOOK_INVALID_HANDLER(hook, typeof fn)
+  if (typeof fn !== 'function') throw new FST_ERR_HOOK_INVALID_HANDLER(hook, fn === null ? null : typeof fn)
```

Comment:
It'd be far easier to read, and have the desired result, to use `Object.prototype.toString.call(fn)` instead.

Reply:
Thanks @jsumners ,
I think now I implemented this in the way you want it to be